Fluid/FluidParser.cs

```diff
@@ -100,7 +100,18 @@ public FluidParser(FluidParserOptions parserOptions)
                     .Or(Call)))
                 .Then(x =>
                 {
-                    return new MemberExpression([x.Item1, .. x.Item2]);
+                    if (x.Item2.Count == 0)
+                    {
+                        return new MemberExpression(x.Item1);
+                    }
+
+                    var list = new List<MemberSegment>(x.Item2.Count + 1);
```

Comment:
I wonder how the actual low-level c# looks when using ` return new MemberExpression([x.Item1, .. x.Item2]);`, the compiler can be quite smart at times and bypass the public API

Reply:
With my latest SDK this:

```c#
List<MemberSegment> memberSegments = [x.Item1, .. x.Item2];
return new MemberExpression(memberSegments);
```

Translates to (according to Rider):

```c#
MemberSegment memberSegment = x.Item1;
List<MemberSegment> collection = x.Item2;
List<MemberSegment> segments = new List<MemberSegment>(1 + collection.Count);
segments.Add(memberSegment);
segments.AddRange((IEnumerable<MemberSegment>) collection);
return new MemberExpression(segments);
```

Reply:
Benchmarks were better with the latest change, but I'll check

Reply:
If I change the csproj target to only have `net8.0`, the low-level c# will be:

```c#
MemberSegment memberSegment = x.Item1;
List<MemberSegment> list = x.Item2;
int num1 = 1 + list.Count;
List<MemberSegment> memberSegmentList = new List<MemberSegment>(num1);
CollectionsMarshal.SetCount<MemberSegment>(memberSegmentList, num1);
Span<MemberSegment> span1 = CollectionsMarshal.AsSpan<MemberSegment>(memberSegmentList);
int index = 0;
span1[index] = memberSegment;
int start = index + 1;
Span<MemberSegment> span2 = CollectionsMarshal.AsSpan<MemberSegment>(list);
span2.CopyTo(span1.Slice(start, span2.Length));
int num2 = start + span2.Length;
return new MemberExpression(memberSegmentList);
```

It's crazy intelligent. I'm not sure if the output will vary based on TFM.

Reply:
Side note, maybe should just drop `net7.0` target.

Reply:
I'll trust the compiler then and not try to be more clever, thanks!